google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/BigQuerySnippets.java

```diff
@@ -663,47 +673,268 @@ public boolean cancelJob(String jobName) {
+   */
+  // [TARGET query(QueryJobConfiguration, JobOption...)]
+  public void runLegacySqlQuery() throws InterruptedException {
+    // [START bigquery_query_legacy]
+    // BigQuery bigquery = BigQueryOptions.getDefaultInstance().getService();
+    String query =
+        "SELECT corpus FROM [bigquery-public-data:samples.shakespeare] GROUP BY corpus;";
     QueryJobConfiguration queryConfig =
+        // To use legacy SQL syntax, set useLegacySql to true.
         QueryJobConfiguration.newBuilder(query).setUseLegacySql(true).build();
+
+    // Print the results.
+    for (FieldValueList row : bigquery.query(queryConfig).iterateAll()) {
+      for (FieldValue val : row) {
+        System.out.printf("%s,", val.toString());
+      }
+      System.out.printf("\n");
+    }
+    // [END bigquery_query_legacy]
+  }
+
+  /**
+   * Example of running a query with Standard SQL explicitly set.
+   */
+  // [TARGET query(QueryJobConfiguration, JobOption...)]
+  public void runStandardSqlQuery() throws InterruptedException {
+    // [START bigquery_query_standard]
+    // BigQuery bigquery = BigQueryOptions.getDefaultInstance().getService();
+    String query =
+        "SELECT corpus FROM `bigquery-public-data.samples.shakespeare` GROUP BY corpus;";
+    QueryJobConfiguration queryConfig =
+        // Note that setUseLegacySql is set to false by default
+        QueryJobConfiguration.newBuilder(query).setUseLegacySql(false).build();
+
+    // Print the results.
+    for (FieldValueList row : bigquery.query(queryConfig).iterateAll()) {
+      for (FieldValue val : row) {
+        System.out.printf("%s,", val.toString());
+      }
+      System.out.printf("\n");
+    }
+    // [END bigquery_query_standard]
+  }
+
+  /**
+   * Example of running a query and saving the results to a table.
+   */
+  // [TARGET query(QueryJobConfiguration, JobOption...)]
+  public void runQueryPermanentTable(String destinationDataset, String destinationTable) throws InterruptedException {
+    // [START bigquery_query_destination_table]
+    // BigQuery bigquery = BigQueryOptions.getDefaultInstance().getService();
+    // String destinationDataset = 'my_destination_dataset';
+    // String destinationTable = 'my_destination_table';
+    String query =
+        "SELECT corpus FROM `bigquery-public-data.samples.shakespeare` GROUP BY corpus;";
+    QueryJobConfiguration queryConfig =
+        // Note that setUseLegacySql is set to false by default
+        QueryJobConfiguration.newBuilder(query)
+        // Save the results of the query to a permanent table. See:
+        // https://cloud.google.com/bigquery/docs/writing-results#permanent-table
+        .setDestinationTable(TableId.of(destinationDataset, destinationTable))
+        // Allow results larger than the maximum response size.
+        // If true, a destination table must be set. See:
+        // https://cloud.google.com/bigquery/docs/writing-results#large-results
+        .setAllowLargeResults(true)
```

Comment:
`allowLargeResults` actually doesn't do anything for standard SQL queries. It is only needed for legacy SQL. We can remove it from this sample.

Perhaps you are planning to use this sample at https://cloud.google.com/bigquery/docs/writing-results#writing_large_results_using_legacy_sql ? Since that sample doesn't set to legacy SQL, it's actually incorrect. I'd be okay dropping the "large results" sample altogether, but if you don't mind making a legacySQL version of this sample, that's probably the best option for user experience.